src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -164,28 +159,17 @@ public int hashCode() {
         return new HashCodeBuilder().append(getModelMetadata()).append(getModelBlob()).toHashCode();
     }
 
-    @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        XContentBuilder xContentBuilder = builder.startObject();
-        if(Strings.hasText(modelID)){
-            builder.field(MODEL_ID, modelID);
-        }
-        builder.field(MODEL_STATE, getModelMetadata().getState().getName());
-        builder.field(MODEL_TIMESTAMP, getModelMetadata().getTimestamp());
-        builder.field(MODEL_DESCRIPTION, getModelMetadata().getDescription());
-        builder.field(MODEL_ERROR, getModelMetadata().getError());
-
-        String base64Model = "";
-        if(getModelBlob() != null){
-            base64Model = Base64.getEncoder().encodeToString(getModelBlob());
-        }
-        builder.field(MODEL_BLOB_PARAMETER, base64Model);
-
-        builder.field(METHOD_PARAMETER_SPACE_TYPE, getModelMetadata().getSpaceType().getValue());
-        builder.field(DIMENSION, getModelMetadata().getDimension());
-        builder.field(KNN_ENGINE, getModelMetadata().getKnnEngine().getName());
-
-        return xContentBuilder.endObject();
+    /**
+     *  Parse source map content into {@link Model} instance.
+     *
+     * @param sourceMap source contents
+     * @param modelID model's identifier
+     * @return model instance
+     */
+    public static Model getModelFromSourceMap(Map<String, Object> sourceMap, @NonNull String modelID) {
+        ModelMetadata modelMetadata = ModelMetadata.getMetadataFromSourceMap(sourceMap);
+        byte[] blob = getModelBlobFromResponse(sourceMap);
```

Comment:
Want to confirm: what happens if these are null?

Reply:
It will not throw any exception. Field value will be empty.